cf-browser/src/main/java/org/eclipse/californium/tools/GUIController.java

```diff
@@ -801,14 +803,22 @@ private void showResponse(Response response) {
 				image = new Image(imgIS);
 			}
 			mediaTypeView.setImage(image);
-			String text = response.getPayloadString();
-			if (format == MediaTypeRegistry.APPLICATION_LINK_FORMAT) {
-				Set<WebLink> webLinks = LinkFormat.parse(text);
-				StringBuilder links = new StringBuilder();
-				for (WebLink link : webLinks) {
-					links.append(link).append(StringUtil.lineSeparator());
+			String text = "";
+			if (format == MediaTypeRegistry.APPLICATION_OCTET_STREAM) {
+				text = StringUtil.byteArray2Hex(payload);
+			} else if (format == MediaTypeRegistry.APPLICATION_CBOR) {
+				CBORObject element = CBORObject.DecodeFromBytes(payload);
```

Comment:
is there any error handling being done at a higher level? If I am not mistaken then this method will throw a runtime exception if the bytes do not represent proper CBOR, right?

Reply:
I will check that. The same would apply to APPLICATION_LINK_FORMAT as well.
I guess, the openjfx protects from crashing on such errors, but sure, generally we may handle that in a better  way.

Reply:
> 09:11:13.647 INFO [GUIController]: Received Response: ACK-2.04   MID=35232, Token=4C4E37DD99458367, OptionSet={"Content-Format":"application/cbor"}, "{"systemstart":1638778241200,"dr".. 311 bytes
> Exception in thread "JavaFX Application Thread" com.upokecenter.cbor.CBORException: Length of 2482461356392279411 is bigger than supported
> 	at com.upokecenter.cbor.CBORReader.ReadStringArrayMap(CBORReader.java:159)
> 	at com.upokecenter.cbor.CBORReader.ReadForFirstByte(CBORReader.java:460)
> 	at com.upokecenter.cbor.CBORReader.ReadInternal(CBORReader.java:151)
> 	at com.upokecenter.cbor.CBORReader.Read(CBORReader.java:123)
> 	at com.upokecenter.cbor.CBORObject.Read(CBORObject.java:3092)
> 	at com.upokecenter.cbor.CBORObject.DecodeFromBytes(CBORObject.java:1084)
> 	at com.upokecenter.cbor.CBORObject.DecodeFromBytes(CBORObject.java:829)
> 	at org.eclipse.californium.tools.GUIController.showResponse(GUIController.java:800)
> 	at org.eclipse.californium.tools.GUIController.access$11(GUIController.java:755)
> 	at org.eclipse.californium.tools.GUIController$ResponsePrinter.lambda$8(GUIController.java:1017)
> 	at com.sun.javafx.application.PlatformImpl.lambda$null$5(PlatformImpl.java:295)
> 	at java.security.AccessController.doPrivileged(Native Method)
> 	at com.sun.javafx.application.PlatformImpl.lambda$runLater$6(PlatformImpl.java:294)
> 	at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:95)
> 	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
> 	at com.sun.glass.ui.gtk.GtkApplication.lambda$null$10(GtkApplication.java:245)
> 	at java.lang.Thread.run(Thread.java:748)

So the exception is catched in the UI loop. 
I redesign it in order to display that as error with the original payload as hex-dump.
(Further PR)

Reply:
See PR #65